third_party/xxhash/cli/xsum_config.h

```diff
@@ -0,0 +1,243 @@
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * You can contact the author at:
+ *   - xxHash homepage: https://www.xxhash.com
+ *   - xxHash source repository: https://github.com/Cyan4973/xxHash
+ */
+
+/*
+ * This contains various configuration parameters and feature detection for
+ * xxhsum.
+ *
+ * Similar to config.h in Autotools, this should be the first header included.
+ */
+
+#ifndef XSUM_CONFIG_H
+#define XSUM_CONFIG_H
+
+
+/* ************************************
+ *  Compiler Options
+ **************************************/
+/*
+ * Disable Visual C's warnings when using the "insecure" CRT functions instead
+ * of the "secure" _s functions.
+ *
+ * These functions are not portable, and aren't necessary if you are using the
+ * original functions properly.
+ */
+#if defined(_MSC_VER) || defined(_WIN32)
+#  ifndef _CRT_SECURE_NO_WARNINGS
+#    define _CRT_SECURE_NO_WARNINGS
+#  endif
+#endif
+
+/* Under Linux at least, pull in the *64 commands */
+#ifndef _LARGEFILE64_SOURCE
+#  define _LARGEFILE64_SOURCE
+#endif
+#ifndef _FILE_OFFSET_BITS
+#  define _FILE_OFFSET_BITS 64
+#endif
+
+/*
+ * So we can use __attribute__((__format__))
+ */
+#ifdef __GNUC__
+#  define XSUM_ATTRIBUTE(x) __attribute__(x)
+#else
+#  define XSUM_ATTRIBUTE(x)
+#endif
+
+#if !defined(_WIN32) && (defined(__unix__) || defined(__unix) || (defined(__APPLE__) && defined(__MACH__)) /* UNIX-like OS */ \
+   || defined(__midipix__) || defined(__VMS))
+#  if (defined(__APPLE__) && defined(__MACH__)) || defined(__SVR4) || defined(_AIX) || defined(__hpux) /* POSIX.1-2001 (SUSv3) conformant */ \
+     || defined(__DragonFly__) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)  /* BSD distros */
+#    define XSUM_PLATFORM_POSIX_VERSION 200112L
+#  else
+#    if defined(__linux__) || defined(__linux)
+#      ifndef _POSIX_C_SOURCE
+#        define _POSIX_C_SOURCE 200112L  /* use feature test macro */
+#      endif
+#    endif
+#include "libc/calls/calls.h"
+#include "libc/calls/weirdtypes.h"
+#include "libc/runtime/pathconf.h"
+#include "libc/runtime/runtime.h"
+#include "libc/runtime/sysconf.h"
+#include "libc/sysv/consts/f.h"
+#include "libc/sysv/consts/fileno.h"
+#include "libc/sysv/consts/o.h"
+#include "libc/sysv/consts/ok.h"
+#include "libc/time/time.h"
+#include "third_party/getopt/getopt.h"
+#include "third_party/musl/crypt.h"
+#include "third_party/musl/lockf.h"  /* declares _POSIX_VERSION */
+#    if defined(_POSIX_VERSION)  /* POSIX compliant */
+#      define XSUM_PLATFORM_POSIX_VERSION _POSIX_VERSION
+#    else
+#      define XSUM_PLATFORM_POSIX_VERSION 0
+#    endif
+#  endif
+#endif
+#if !defined(XSUM_PLATFORM_POSIX_VERSION)
+#  define XSUM_PLATFORM_POSIX_VERSION -1
+#endif
+
+/**
+ * 
+ * fzakaria: remove this
+ * 
+ * #if !defined(S_ISREG)
+ * #  define S_ISREG(x) (((x) & S_IFMT) == S_IFREG)
+ * #endif
+ */
```

Comment:
I had to hand-edit the source here -- this macro becomes redefined by the libc the way the includes work.